### DISCUSSION TITLE
Use new minor version for breaking changes

### DISCUSSION THREAD

**Discussion by gisostallenberg** [rectorphp/rector#8059] [General]
To allow related packages to have proper version constraints I think any breaking change should increment the minor version, while in pre v1 version.
Release https://github.com/rectorphp/rector/releases/tag/0.17.3 introduced a break by removing findParentType. Due to this `driftingly/rector-laravel` is not working properly with this version (see https://github.com/driftingly/rector-laravel/issues/121).

It could have a version constraint `rector/rector:^0.17` and not break if rector would have incremented to 0.18.0.

**Comment by TomasVotruba** [comment]
Good point :pray: , I'll try to think about it in next releases.

The linked parent node removal is promoted since Rector 0.17: https://github.com/rectorphp/rector/releases/tag/0.17.0
We wanted to split it into more versions, so the removal is step-by-step and not all at once.

The best way to keep up is to follow release notes on rector/rector repository.

**Reply by Wirone** [reply]
@TomasVotruba as you can see @gisostallenberg was aware of changes and even linked release notes, so suggesting to follow them is pretty not accurate 😉. I understand Rector is still in so-called "initial development" (`0.*`), but it would be really helpful if breaking changes were introduced only in minor releases (even though SemVer clearly states that for initial development everything may change at any point). Especially that there is not enough information about migrating code to the new API - for example `BetterNodeFinder::findParentType()` was deprecated with `@deprecated Make use of child nodes instead` tip, which I find very laconic and not helpful. Currently I'm trying to update Rector and have hard time refactoring custom rules (`0.17.1` → `0.17.14`).

**Reply by TomasVotruba** [reply]
Noted, I'll be more careful about this next time. Thanks for feedback :pray:
